pkg/ingester/ingester.go

```diff
@@ -3189,11 +3178,30 @@ func (i *Ingester) Push(ctx context.Context, req *mimirpb.WriteRequest) (*mimirp
 
 func (i *Ingester) handlePushError(err error) error {
 	if i.cfg.ReturnOnlyGRPCErrors {
-		return handlePushErrorWithGRPC(err)
+		return handlePushError(err)
 	}
 	return handlePushErrorWithHTTPGRPC(err)
 }
 
+func (i *Ingester) handleReadError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	if errors.Is(err, context.Canceled) {
+		return err
+	}
+
+	if errors.Is(err, context.DeadlineExceeded) {
+		return err
+	}
```

Comment:
[Nit] Simplification:
```suggestion
	if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
		return err
	}
```

Reply:
Thank you, I will apply that.